core/src/main/scala/chisel3/Const.scala

```diff
@@ -0,0 +1,20 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3
+
+import chisel3._
+import chisel3.internal.{requireIsChiselType, Builder}
+
+/** Create a constant type in FIRRTL, which is guaranteed to take a single
+  * constant value.
+  */
+object Const {
+  def apply[T <: Data](source: => T): T = {
+    val prevId = Builder.idGen.value
+    val data = source // should only evaluate source once
+    requireIsChiselType(data)
+    val ret = if (!data.mustClone(prevId)) data else data.cloneType.asInstanceOf[T]
+    ret.isConst = true
+    ret
```

Comment:
Consider using mutation here. I've gotten used to the following structure and find that it is more readable:

```suggestion
    var data = source // should only evaluate source once
    requireIsChiselType(data)
    if (data.mustClone(prevId)
      data = data.cloneType.asInstanceOf[T]
    data.isConst = true
    data
```

Not blocking, only for your consideration.